src/sagemaker/model.py

```diff
@@ -1696,8 +1696,7 @@ def deploy(
                 instance_type=instance_type,
                 accelerator_type=accelerator_type,
                 tags=tags,
-                serverless_inference_config=serverless_inference_config,
-                **kwargs,
+                serverless_inference_config=serverless_inference_config
```

Comment:
oof, this looks like a backward incompatible change, are you sure we can simply do that?

We have to explicitly list the arguments that are eligible to be passed here.